src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs

```diff
@@ -0,0 +1,124 @@
+/// Contains a task which allows to run [SpecFlow](http://www.specflow.org/) tests with SpecFlow v2.4+.
+[<RequireQualifiedAccess>]
+module Fake.DotNet.Testing.SpecFlowNext
+
+open Fake.Core
+open Fake.IO
+open Fake.IO.Globbing
+open Fake.IO.FileSystemOperators
+open System.IO
+open System.Text
+
+type SubCommand =
+    | GenerateAll
+    | StepDefinitionReport
+    | NUnitExecutionReport
+    | MsTestExecutionReport
+
+    override x.ToString () =
+        match x with
+        | GenerateAll -> "GenerateAll"
+        | StepDefinitionReport -> "StepDefinitionReport"
+        | NUnitExecutionReport -> "NUnitExecutionReport"
+        | MsTestExecutionReport -> "MsTestExecutionReport"
+
+/// SpecFlow execution parameter type.
+type SpecFlowParams = { 
+    SubCommand:         SubCommand
+    ProjectFile:        string
+    ToolName:           string
+    ToolPath:           string
+    WorkingDir:         string
+    BinFolder:          string option
+    OutputFile:         string option
+    XmlTestResultFile:  string option
+    TestOutputFile:     string option
+    FeatureLanguage:    string option
+    Verbose:            bool
+    ForceRegeneration:  bool
+    XsltFile:           string option
+}
+
+let private toolname = "specflow.exe"
+let private currentDirectory = Directory.GetCurrentDirectory ()
+
+/// SpecFlow default execution parameters.
+let private SpecFlowDefaults = { 
+    SubCommand =        GenerateAll
+    ProjectFile =       null
+    ToolName =          toolname
+    ToolPath =          Tools.findToolFolderInSubPath toolname (currentDirectory </> "tools" </> "SpecFlow")
+    WorkingDir =        null
+    BinFolder =         None
+    OutputFile =        None
+    XmlTestResultFile = None
+    TestOutputFile =    None
+    FeatureLanguage =   None
+    Verbose =           false
+    ForceRegeneration = false
+    XsltFile =          None
+}
+
+let internal createProcess setParams =
+    let parameters = setParams SpecFlowDefaults
+    let tool = parameters.ToolPath </> parameters.ToolName
+
+    let yieldIfSome paramName value =
+        seq { match value with
+              | Some v ->
+                yield sprintf "--%s" paramName
+                yield v
+              | _ -> () }
+
+    let args = 
+        [
+            yield parameters.SubCommand |> string
+            
+            if not (isNull parameters.ProjectFile) then
```

Comment:
@magicmonty I noticed while writing the test that `null` leads to an ugly exception. Is this parameter required? Can we give a good default>

Reply:
Which one gives an exception? The SubCommand or the ProjectFile?
Both are needed. For the SubCommand the default should be "generateall" 
The Project file is needed and thus must not be null. This is the reason I added options on the SpecFlowNext variant

Reply:
Ok if it is needed can you please remove the nullcheck I added and make sure that user gets a proper error message if the parameter is not given?

Reply:
@matthid 
I removed the null check and adjusted the API a bit. The ProjectFile now needs to be provided on "run". I have opened a new PR for this, as I cannot push to this branch